### DISCUSSION TITLE
Does HTTPS require client auth certificate?

### DISCUSSION THREAD

**Discussion by damccull** [ravendb/ravendb#14966] [Q&A] [answered]
Does using HTTPS require a client auth cert or is there another way to authenticate to the server? Also, does /not/ using https have a way to require the user to authenicate or is it just automatically open?

**Comment by ayende** [comment, marked as answer]
Yes, we support only client auth certificate.
If you don't use HTTPS, you don't have an authentication method.

Note that we have the Let's Encrypt automatic setup, see: 
https://www.youtube.com/watch?v=K-2iZ_lJVag&t=119s

**Comment by roqvist** [comment]
I realize this is a minor detail, and I'm definitely an edge case, but an advanced development/unprotected configuration where only server certificate can be configured, would be nice in certain cases.

I'm running an unprotected Raven in development/HTTP mode along with a couple of other services on HTTPS, on the same host. In this case you can not access Studio in the web browser due to HSTS; the browser will automatically force HTTPS. You need to clear out any HSTS cached settings, or visit through a cookie-less anonymous tab, to allow a mix of HTTP and HTTPS.

The obvious solution with client certificates doesn't work in my case, as I have integrated a couple of very limited embedded devices that simply cannot have their own client certificates.

**Reply by ayende** [reply]
Why can't the env use client certs?

Our thinking is that the network is hostile:
https://ayende.com/blog/189729-C/the-network-is-always-hostile

It doesn't make sense to be encrypted and not have certificate.

**Reply by ayende** [reply]
If you *really* need that scenario, put an HTTPS proxy in front of RavenDB, that would solve your problem, but would explicitly not be setting up RavenDB in a secured fashion.

**Reply by roqvist** [reply]
I'm with you on the security, I really like the approach with client certificates. It was just a thought to at least make such a configuration possible (HTTPS server certificate without authentication on clients) for specific use cases. Obviously still considered an insecure/development setup.

**Reply by roqvist** [reply]
Good idea with the proxy. Probably better than to change anything with Raven since this is such an edge case.
